### DISCUSSION TITLE
How to properly troubleshoot/profile Volar performance issues?

### DISCUSSION THREAD

**Discussion by fabis94** [vuejs/language-tools#3504] [Q&A]
Volar is running really slow in a project of mine - VSCode Save actions take more than 6 seconds, intellisense is slow etc., so I'd like to troubleshoot/profile Volar to see what causes the slowdown.

But there don't appear to be any up-to-date resources regarding how to do this properly. I found this: https://github.com/vuejs/language-tools/discussions/413

It seems to be a bit outdated - It references https://github.com/vuejs/vetur/blob/master/.github/PERF_ISSUE.md#profiling which is not even for Volar, but Vetur - but it does kinda work. The problem is that the profiling traces can't be opened in the browser, possibly because they're too huge? I can never get it to get past "Processing profile" in dev tools.

I feel like performance issues can be pretty common in Volar, so there should be some effort to try to build or improve some troubleshooting tools/scripts that would make it easier to figure out what causes slowdowns.

**Comment by Suniron** [comment]
> Volar is running really slow in a project of mine - VSCode Save actions take more than 6 seconds, intellisense is slow etc., so I'd like to troubleshoot/profile Volar to see what causes the slowdown.
> 
> But there don't appear to be any up-to-date resources regarding how to do this properly. I found this: https://github.com/vuejs/language-tools/discussions/413
> 
> It seems to be a bit outdated - It references https://github.com/vuejs/vetur/blob/master/.github/PERF_ISSUE.md#profiling which is not even for Volar, but Vetur - but it does kinda work. The problem is that the profiling traces can't be opened in the browser, possibly because they're too huge? I can never get it to get past "Processing profile" in dev tools.
> 
> I feel like performance issues can be pretty common in Volar, so there should be some effort to try to build or improve some troubleshooting tools/scripts that would make it easier to figure out what causes slowdowns.

Hey,

Did you find anything about that? I have some performance issues and volar crashes..
